src/actions/motion.ts

```diff
@@ -120,69 +120,144 @@ class MoveDownByScreenLine extends MoveByScreenLine {
+       * only if the position we were previously was not at the desired column
        */
-      let start = Position.FromVSCodePosition(vimState.editor.selection.start);
-      if (
-        (this.movementType === 'down' && position.line > start.line) ||
-        (this.movementType === 'up' && position.line < prevLine)
-      ) {
-        await vscode.commands.executeCommand('cursorMove', {
-          to: 'left',
-          select: true,
-          by: 'character',
-          value: 1,
-        });
+      // Move VSCode cursor to 'position'
+      let active = vimState.editor.selection.active;
+      if (!active.isEqual(position)) {
+        // We need to use while loops here because when inside a wrapped line if you try to move
+        // to a position that is on a different "visual" line VSCode won't move directly to the
+        // position you want, first it moves to the end/start of the current "visual" line, depending
+        // if you're moving right or left, respectively.
+        while (active.isBefore(position) && active.line <= position.line) {
+          await vscode.commands.executeCommand('cursorMove', {
+            to: 'right',
+            select: true,
+            by: 'character',
+            value: Math.abs(position.character - active.character),
+          });
+          active = vimState.editor.selection.active;
+        }
+        while (position.isBefore(active) && active.line >= position.line) {
+          await vscode.commands.executeCommand('cursorMove', {
+            to: 'left',
+            select: true,
+            by: 'character',
+            value: Math.abs(active.character - position.character),
+          });
+          active = vimState.editor.selection.active;
+        }
       }
     }
 
+    const previousLineColumn = vimState.cursorStopPosition.character;
+
     await vscode.commands.executeCommand('cursorMove', {
       to: this.movementType,
       select: vimState.currentMode !== Mode.Normal,
       by: this.by,
-      value: this.value,
+      value: this.value * count,
     });
 
+    // Set active position to the desired column
+    let activePos = Position.FromVSCodePosition(vimState.editor.selection.active);
+    if (activePos.character < desiredColumn && activePos.line !== prevLine) {
+      activePos = activePos.withColumn(desiredColumn);
+    } else if (previousLineColumn < desiredColumn) {
+      // If the active position is after desiredColumn it means we moved to a wrapped line.
+      // In that case if the position on previous line is less then desiredColumn we need to
+      // move right by the difference of desiredColumn and the previous line column.
+      const indent = vscode.workspace.getConfiguration('editor').get('wrappingIndent');
+      const indentSpace = TextEditor.getFirstNonWhitespaceCharOnLine(activePos.line).character;
+      const tabSize = vscode.workspace.getConfiguration('editor').get<number>('tabSize')!;
```

Comment:
I believe you want to use `vimState.editor.options.tabSize` instead

Reply:
Yes you're correct, nice catch!